### PR TITLE
New extension: load_tex_macros.

### PR DESCRIPTION
**Pull request by ketch**
Just loads a tex file whenever a notebook is loaded, then re-runs mathjax.  Useful if you have several notebooks that use a common set of latex macros, so you don't have to copy the macros to each notebook.

Essentially all credit for this goes to @jfbercher who practically wrote the extension for me; it's based on  one of the functions from his `latex_env` extension.

### CODE REVIEW

**Comment by ketch** [comment]
It looks like all the Travis jobs that actually ran passed.  There are some failures, but those jobs seem never even to have started.

What can I do to help git this PR merged?

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/load_tex_macros.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Type: IPython Notebook Extension
```
Comment (jcb91) [inline src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/load_tex_macros.yaml]: Could you change this to `Jupyter Notebook Extension`? Thanks!

**Comment by jcb91** [comment]
Travis builds restarted, but they aren't going to check anything relevant for this all-js extension anyway, tbh, so not a problem to ignore them here if they fail...

**Comment by ketch** [comment]
@jcb91 Thanks for the review.  I changed "IPython" to "Jupyter".

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+define(function(require, exports, module) {
+    var Jupyter = require('base/js/namespace');
+
+    function loadLatexUserDefs() { 
+        $.get('latexdefs.tex').done(function(data) {
+            data = data.replace(/^/gm, '\$\$\$').replace(/$/gm, '\$\$\$');
+            if ($('#latexdefs').length > 0) $('#latexdefs').remove();
+            $('body').append($('<div/>').attr('id', 'latexdefs').text(data));
+            console.log('latex_envs: loaded user LaTeX definitions latexdefs.tex');
+            onMarkdownCellRendering();
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js]: I am not sure it works because function `onMarkdownCellRendering()` is not defined. It is defined in `latex_envs` and is global (which is not a good practice indeed), but in the absence of `latex_envs`  extension, this will not work.

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/load_tex_macros.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Type: Jupyter Notebook Extension
+Name: Load TeX macros
+Description: This extension automatically loads a set of latex commands from the file latexdefs.tex when a notebook is opened.
+Link: readme.md
+Icon: icon.png
+Main: main.js
+Compatibility: 4.x
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/load_tex_macros.yaml]: add 5.X in `Compatibility: 4.x, 5.x`

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        }).fail(function() {
+            console.log('load_tex_macros: failed to load user LaTeX definitions latexdefs.tex')
+        });
+    }
+
+    function rerenderMaths() { // probably something like that
+            MathJax.Hub.Queue(
+              ["PreProcess", MathJax.Hub], ["Reprocess", MathJax.Hub]
+            );
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js]: I think 
```
MathJax.Hub.Queue(
  ["resetEquationNumbers",MathJax.InputJax.TeX],
  ["PreProcess",MathJax.Hub],
  ["Reprocess",MathJax.Hub]
);
``` 
could be better (resets equation numbers before reprocessing avoid an issue with labels in equations that cannot be rendered several times without reset.

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/readme.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Usage
+=====
+
+Simply put your latex macros in a file named latexdefs.tex, in the same directory as your notebook.
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/readme.md]: Perhaps "This is mainly derived from [jupyter_latex_envs](https://pypi.python.org/pypi/jupyter_latex_envs), following the discussion [here](https://github.com/jupyter/notebook/issues/3358)".

Comment (jfbercher) [reply src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/readme.md]: Sorry! didn't read the mention at the beginning. Thanks.

File: src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js (hunk 2 above)
Comment (ketch) [reply src/jupyter_contrib_nbextensions/nbextensions/load_tex_macros/main.js]: Hmmm, somehow it works for me even with `latex_envs` disabled.

**Comment by ketch** [comment]
Okay, I've incorporated all the suggestions form @jfbercher .  I think this is ready to go now.

**Comment by jfbercher** [comment]
Nice and useful. Thanks @ketch!
